node/pegnet/bank.go

```diff
@@ -0,0 +1,150 @@
+package pegnet
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/pegnet/pegnet/modules/conversions"
+)
+
+// BankEntry is for managing the PEG Bank.
+// The bank is the amount of PEG allowed to be converted into every block.
+// This table should be used for bank management purposes.
+//
+// Originally the bank was thought of in order to allow conversions out of PEG
+// to positively impact the bank, allowing more conversions in.
+// This would require the supply to be under the supply curve, which we are
+// no where close too.
+//
+// So now the bank has a different control. The bank will be controlled
+// by other mechanisms. Each height will have the following struct data stored
+//
+// There is extra information provided for informational purposes. The only
+// true value needed here is likely the BankAmount and possible the BankUsed.
+// The PEGRequested shows the demand for PEG at a high level.
+type BankEntry struct {
+	// Each height will have a bank entry.
+	Height int32
+	// BankAmount is the total amount of PEG allowed to be converted into
+	// for the given height.
+	BankAmount int64 // units are PEG
+	// BankUsed is some additional information to detail how much of the bank
+	// was consumed.
+	BankUsed int64 // units are PEG
+	// PEGRequested is the total amount of PEG requested
+	PEGRequested int64 // units are PEG
+}
+
+const createTableBank = `CREATE TABLE IF NOT EXISTS "pn_bank" (
+	"height" INTEGER PRIMARY KEY,
+	"bank_amount" INTEGER NOT NULL DEFAULT 0,
+	"bank_used" INTEGER NOT NULL DEFAULT 0,
+	"total_requested" INTEGER NOT NULL DEFAULT 0,
+	
+	UNIQUE("height")
+);
+`
+
+const (
+	// Bank Properties and constants
+
+	// BankBaseAmount is the minimum the bank size will ever be.
+	// Any bank growth will be ontop of this value.
+	BankBaseAmount = conversions.PerBlock
+
+	// BankGrowth
+	BankGrowthAmount = 500 * 1e8
+	BankDecayAmount  = BankGrowthAmount
+
+	// BankMaxLimit is the maximum size of the Bank for any given block
+	// 288 blocks of growth
+	BankMaxLimit = BankBaseAmount + (288 * BankGrowthAmount)
+)
+
+// CreateTableBank is used to expose this table for unit tests
+func (p *Pegnet) CreateTableBank() error {
```

Comment:
It's just used in unit tests. I can't access private fields in the unit tests, and our tables are unexported constants. If we change all the table strings to exported constants, then I don't need an exported function.

Reply:
Ah, it looks like you're putting all your unit tests into their own package (pegnet_test). Creating exported functions just for unit testing seems like a waste, I personally prefer to make the unit tests part of the normal package, which seems to be common practice.

Reply:
Ah,  I've been using _test for awhile now. That would allow me to not need to export the functions. I think we can adjust this in another PR, as there is more than 1 test that does this.